app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java

```diff
@@ -117,6 +123,87 @@ protected void onCreate(Bundle savedInstanceState) {
         // copy clash template configuration
         copyRawResourceToDir(R.raw.clash_config, Constants.getClashTemplatePath(), true);
 
+        final TrojanShareLink trojanShareLink = new TrojanShareLink();
+
+        shareLinkText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) { }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) { }
```

Comment:
Why empty method body?

Reply:
`TextWatcher` defines 3 methods, so I have to implement all the interface members.